pkg/golinters/depguard.go

```diff
@@ -63,47 +129,31 @@ func NewDepguard() *goanalysis.Linter {
 	).WithContextSetter(func(lintCtx *linter.Context) {
 		dgSettings := &lintCtx.Settings().Depguard
 		analyzer.Run = func(pass *analysis.Pass) (interface{}, error) {
-			prog := goanalysis.MakeFakeLoaderProgram(pass)
-			dg := &depguard.Depguard{
-				Packages:      dgSettings.Packages,
-				IncludeGoRoot: dgSettings.IncludeGoRoot,
-			}
-			if err := setDepguardListType(dg, lintCtx); err != nil {
+			parsedDgSettings, err := parseDepguardSettings(dgSettings)
+			if err != nil {
 				return nil, err
 			}
-			setupDepguardPackages(dg, lintCtx)
 
 			loadConfig := &loader.Config{
 				Cwd:   "",  // fallbacked to os.Getcwd
 				Build: nil, // fallbacked to build.Default
 			}
-			issues, err := dg.Run(loadConfig, prog)
-			if err != nil {
-				return nil, err
-			}
-			if len(issues) == 0 {
-				return nil, nil
-			}
```

Comment:
Note that this is taken out so that we don't accidentally short-circuit additional guards.